syntax/vm/vm.go

```diff
@@ -475,22 +470,13 @@ func NewScope(variables map[string]interface{}) *Scope {
 	}
 }
 
-func NewScopeWithParent(parent *Scope, variables map[string]interface{}) *Scope {
-	return &Scope{
-		Parent:    parent,
-		Variables: variables,
-	}
-}
-
-// Lookup looks up a named identifier from the scope, all of the scope's
-// parents, and the stdlib.
+// Lookup looks up a named identifier from the scope and the stdlib.
 func (s *Scope) Lookup(name string) (interface{}, bool) {
 	// Traverse the scope first, then fall back to stdlib.
-	for s != nil {
+	if s != nil {
```

Comment:
Is this still a traversal? Comment may need some cleanup

Reply:
no not anymore, will update